### PR TITLE
Add new field for SimulationControl: `DoHVACSizingSimulationforSizingPeriods` and `MaximumNumberofHVACSizingSimulationPasses`

### LINKED ISSUE #192: Add new fields for SimulationControl: DoHVACSizingSimulationforSizingPeriods and MaximumNumberofHVACSizingSimulationPasses
Enhancement Request
--------------

OS SDK 3.0.0 added two new fields for SimulationControl: `DoHVACSizingSimulationforSizingPeriods` and `MaximumNumberofHVACSizingSimulationPasses`

cf https://github.com/NREL/OpenStudio/pull/3897/files

## Possible Implementation

Add them to the GUI on the Settings Tab > Simulation Control

### PR DESCRIPTION
**Pull request by jmarrec**
Fix #192 - Add new field for SimulationControl: `DoHVACSizingSimulationforSizingPeriods` and `MaximumNumberofHVACSizingSimulationPasses`

### CODE REVIEW

**Comment by jmarrec** [comment]
New fields at the bottom:

image

File: developer/doc/ReleaseNotes/OpenStudioApplication_Release_Notes_1_1_0.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+These are specific to the Application. Please refer to the release notes for OpenStudio SDK for 3.1.0 release to see what has changed with the SDK.
+
+* Updated [OpenStudio SDK](https://github.com/NREL/OpenStudio) from v3.0.0 to v3.1.0
+
+* Updated [EnergyPlus](https://github.com/NREL/EnergyPlus) from v9.3.0 to v9.4.0
+
+* [#185](https://github.com/NREL/OpenStudioApplication/pull/185) - Support WaterCooled VRFs in OSApp
+
+* [#193](https://github.com/NREL/OpenStudioApplication/pull/193) - Add new fields for SimulationControl: `DoHVACSizingSimulationforSizingPeriods` and `MaximumNumberofHVACSizingSimulationPasses`
```
Comment (jmarrec) [inline developer/doc/ReleaseNotes/OpenStudioApplication_Release_Notes_1_1_0.md:68]: Add a draft release note, and write the two new features in the works currently.

File: src/openstudio_lib/SimSettingsView.cpp [hunk 2]
```diff
@@ -140,6 +140,9 @@ SimSettingsView::SimSettingsView(bool isIP,
   m_loadsConvergenceToleranceValue(nullptr),
   m_temperatureConvergenceToleranceValue(nullptr),
   m_solarDistribution(nullptr),
+  m_doHVACSizingSimulationforSizingPeriods(nullptr),
+  m_maximumNumberofHVACSizingSimulationPasses(nullptr),
```
Comment (jmarrec) [inline src/openstudio_lib/SimSettingsView.cpp:144]: Ctor initialize new fields in the right place to match hpp

File: src/openstudio_lib/SimSettingsView.cpp [hunk 3]
```diff
@@ -563,6 +566,15 @@ QWidget * SimSettingsView::createSimulationControlWidget()
...
 
+  row = row + 2;
+  spacerItem = new QSpacerItem(1,SPACERITEM_HEIGHT,QSizePolicy::Fixed,QSizePolicy::Fixed);
+  gridLayout->addItem(spacerItem,row++,0);
+  col = 0;
+
+  addField(gridLayout,row,col,"Do HVAC Sizing Simulation for Sizing Periods",m_doHVACSizingSimulationforSizingPeriods);
+  col = col + 2;
+  addField(gridLayout,row,col,"Maximum Number of HVAC Sizing Simulation Passes",m_maximumNumberofHVACSizingSimulationPasses);
+
```
Comment (jmarrec) [inline src/openstudio_lib/SimSettingsView.cpp:577]: Add an switch and an integer edit for the two new fields

File: src/openstudio_lib/SimSettingsView.cpp [hunk 4]
```diff
@@ -1374,6 +1386,22 @@ void SimSettingsView::attachSimulationControl()
...
+  m_maximumNumberofHVACSizingSimulationPasses->bind(
+      *m_simulationControl,
+      IntGetter(std::bind(&model::SimulationControl::maximumNumberofHVACSizingSimulationPasses,m_simulationControl.get_ptr())),
+      boost::optional<IntSetter>(std::bind(&model::SimulationControl::setMaximumNumberofHVACSizingSimulationPasses,m_simulationControl.get_ptr(),std::placeholders::_1)),
+      boost::optional<NoFailAction>(std::bind(&model::SimulationControl::resetMaximumNumberofHVACSizingSimulationPasses,m_simulationControl.get_ptr())),
+      boost::none,
+      boost::none,
+      boost::optional<BasicQuery>(std::bind(&model::SimulationControl::isMaximumNumberofHVACSizingSimulationPassesDefaulted,m_simulationControl.get_ptr())));
+
```
Comment (jmarrec) [inline src/openstudio_lib/SimSettingsView.cpp:1404]: Bind the two new fields.

File: src/openstudio_lib/SimSettingsView.cpp [hunk 5]
```diff
@@ -1950,6 +1978,8 @@ void SimSettingsView::detachSimulationControl()
   m_loadsConvergenceToleranceValue->unbind();
   m_temperatureConvergenceToleranceValue->unbind();
   m_solarDistribution->unbind();
+  m_doHVACSizingSimulationforSizingPeriods->unbind();
+  m_maximumNumberofHVACSizingSimulationPasses->unbind();
```
Comment (jmarrec) [inline src/openstudio_lib/SimSettingsView.cpp:1982]: Unbind them on detach

File: src/openstudio_lib/SimSettingsView.hpp [hunk 6]
```diff
@@ -225,6 +225,8 @@ class SimSettingsView : public QWidget, public Nano::Observer
   OSQuantityEdit2 * m_loadsConvergenceToleranceValue;
   OSQuantityEdit2 * m_temperatureConvergenceToleranceValue;
   OSComboBox2 * m_solarDistribution;
+  OSSwitch2 * m_doHVACSizingSimulationforSizingPeriods;
+  OSIntegerEdit2 * m_maximumNumberofHVACSizingSimulationPasses;
```
Comment (jmarrec) [inline src/openstudio_lib/SimSettingsView.hpp:229]: One switch for the YES/NO one, and one integer edit.

File: developer/doc/ReleaseNotes/OpenStudioApplication_Release_Notes_1_1_0.md (hunk 1 above)
Comment (macumber) [reply developer/doc/ReleaseNotes/OpenStudioApplication_Release_Notes_1_1_0.md:68]: Cool

File: src/openstudio_lib/SimSettingsView.cpp (hunk 2 above)
Comment (macumber) [reply src/openstudio_lib/SimSettingsView.cpp:144]: We should turn on static analysis to check this, http://cppcheck.sourceforge.net/

**Comment by macumber** [comment]
Want me to merge?

**Pull request merged**